### PR TITLE
[Homework #19] - React Hooks by Rodrigo Collins

### PR DESCRIPTION
**Pull request by RodrigoCollins**

### CODE REVIEW

File: homeworks/rodrigo-collins/19-react-hooks/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+  const addTodo = () => {
+    const trimmedTask = task.trim();
+
+    if (!validateTodoText(trimmedTask)) {
+      return;
+    }
+
+    if (todos.some((todo) => todo.text === trimmedTask)) {
+      alert('To-Do already exists');
```
Comment (ababenko-solvd) [inline homeworks/rodrigo-collins/19-react-hooks/src/App.tsx:49]: This error goes also if you save todo with the same text - but it should not

File: homeworks/rodrigo-collins/19-react-hooks/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      return false;
+    }
+
+    return true;
+  };
+
+  const addTodo = () => {
+    const trimmedTask = task.trim();
+
+    if (!validateTodoText(trimmedTask)) {
```
Comment (ababenko-solvd) [inline homeworks/rodrigo-collins/19-react-hooks/src/App.tsx:44]: Hint:
you can write 1-line conditions in such way - will look better and shorter:

if (!validateTodoText(trimmedTask)) return;

File: homeworks/rodrigo-collins/19-react-hooks/src/components/CreateTodoInput.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    inputRef: React.RefObject<HTMLInputElement>;
+}
+
+const CreateTodoInput: React.FC<CreateTodoInputProps> = ({ task, setTask, inputRef }) => {
+    return (
+        <input
+            ref={inputRef}
+            className='todo-input'
+            type="text"
+            placeholder="Create Todo-Task"
```
Comment (ababenko-solvd) [inline homeworks/rodrigo-collins/19-react-hooks/src/components/CreateTodoInput.tsx:15]: Placeholder should not be shown, when the input is focused

File: homeworks/rodrigo-collins/19-react-hooks/src/components/TodoItem.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import { useState } from 'react';
+import deleteImg from '../assets/delete 1.png'
+import editImg from '../assets/write 1.png'
+import checkImg from '../assets/check.svg'
+
+interface Todo {
```
Comment (ababenko-solvd) [inline homeworks/rodrigo-collins/19-react-hooks/src/components/TodoItem.tsx:6]: I see nice progress with types!
Have some more advices.

1) "interface"  is for types inheritance and extending.
For simple types use "type"
type Todo = {
    id: number;
    text: string;
    completed: boolean;
}

2) Here you have usage of the same types in different components and you are redeclaring them again - this is wrong!
- declare them once(in component or in separate file with types) with export
- import that type to another component(s) and use

then if there would be type changes (they frequently appear) - you will update type in one place

**Review by ababenko-solvd — CHANGES_REQUESTED** [review]

File: homeworks/rodrigo-collins/19-react-hooks/src/components/TodoList.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+                    toggleTodo={toggleTodo}
+                    deleteTodo={deleteTodo}
+                    editTodo={editTodo}
+                />
+            ))}
+        </>
+    );
+}
+
+export default TodoList;
```
Comment (ababenko-solvd) [inline homeworks/rodrigo-collins/19-react-hooks/src/components/TodoList.tsx:32]: Complementary tasks for you, if you have time and motivation:

1) Add ability to add new todo not only by clicking the btn, but also by pressing "Enter"
2) Stop using Alerts for errors - add some UI based solution

**Review by ababenko-solvd — CHANGES_REQUESTED** [review]

File: homeworks/rodrigo-collins/19-react-hooks/src/components/TodoItem.tsx (hunk 4 above)
Comment (RodrigoCollins) [reply homeworks/rodrigo-collins/19-react-hooks/src/components/TodoItem.tsx:6]: Done!

File: homeworks/rodrigo-collins/19-react-hooks/src/App.tsx (hunk 2 above)
Comment (RodrigoCollins) [reply homeworks/rodrigo-collins/19-react-hooks/src/App.tsx]: done

File: homeworks/rodrigo-collins/19-react-hooks/src/components/CreateTodoInput.tsx (hunk 3 above)
Comment (RodrigoCollins) [reply homeworks/rodrigo-collins/19-react-hooks/src/components/CreateTodoInput.tsx]: done

**Review by ababenko-solvd — APPROVED** [review]

**Pull request merged**